informer/collector.go

```diff
@@ -0,0 +1,54 @@
+package informer
```

Comment:
I don't really like this in the same package, can you guys give me some advice how I can move this to a different package and use the private method `fillCache` below? Only way I could think of was to make `fillCache` public and pass the informer through to the collector but that doesn't seem much better.

Reply:
This is totally fine with me.

Reply:
I don't see anything wrong with this sitting in informer package either. But I would create a separate struct for the collector. Still has access to private methods of Informer.

Reply:
Can you explain what you would put in the struct? I don't quite follow.

Reply:
The separate struct does not make sense to me. It would mean we have another object and another constructor and we have to wire all this. To me it is fine as it is. I would just like to remove the dependency to `fillCache` and implement the required functionality separately.

Reply:
I meant a struct with `Describe` and `Collect`, but I didn't realize it uses `watcher`. So scratch that.